syft/execution/translation/torchscript.py

```diff
@@ -0,0 +1,36 @@
+from torch import jit
+from syft.execution.placeholder import PlaceHolder
+from syft.execution.translation.abstract import AbstractPlanTranslator
+
+
+class PlanTranslatorTorchscript(AbstractPlanTranslator):
+    """Performs translation from 'list of ops' Plan into torchscript Plan"""
+
+    def __init__(self, plan):
+        super().__init__(plan)
+
+    def translate(self):
+        plan = self.plan
+
+        args_shape = plan.get_args_shape()
+        args = PlaceHolder.create_placeholders(args_shape)
+
+        # Temporarily remove reference to original function
+        tmp_forward = plan.forward
+        plan.forward = None
+
+        # To avoid storing Plan state tensors in torchscript, they will be send as parameters
+        plan_params = plan.parameters()
```

Comment:
nit: this variable is not used.
Also, I didn't get your comment just above. Is it related to something torchscript-specific?

Reply:
> Is it related to something torchscript-specific?

Yes, jit.trace() fails if tensors with require_grad=True are used in operations but created outside of traced scope, like this:
```
t = torch.randn(3,3, requires_grad=True)

def my_func(x):
   x = x + t * 2
   return t

torch.jit.trace(my_func, (torch.ones(3, 3),)) #  <== Fails because t is created somewhere outside of my_func
```

We either need to register such tensors via nn.Module (this will store tensor inside torchscript, which we want to avoid here) or pass them as an argument into traced function:
```
t = torch.randn(3,3, requires_grad=True)

def my_func(x, t):
   x = x + t * 2
   return t

torch.jit.trace(my_func, (torch.ones(3, 3), t)) #  <== works
```

Reply:
Fixed `plan_params` 👍

Reply:
Hum, and it wouldn't work with the `state.read()`, I guess